### ISSUE TITLE
Error with moon-router and monx

### ISSUE THREAD

**Issue by rogierverbrugge** [KingPixil/moon#91]
Still playing with moon but having trouble with combining moon, router and monx. The following code fails if the user wants to increment when he has visited that route before.

parentNode not found displays the console.

Any idea what's going on?

```js
import Moon from 'moonjs/dist/moon.js'
import Router from 'moon-router'
import Monx from 'monx'

require('./bootstrap');

Moon.use(Router)
Moon.use(Monx)

const store = new Monx({
  state: {
    count: 1
  },
  actions: {
    increment: function(state, payload) {
      state.count += payload;
    }
  }
});

var router = new Router({
  default: "/",
  map: {
    "/": "Root",
    "/hello": "Hello"
  }
});

Moon.component("Root", {
  template: `<div>
    <h1>Welcome to "/"</h1>
    <router-link to="/hello">To /hello</router-link>
  </div>`
});

Moon.component("Hello", {
  template: `<div>
    <h1>{{store.state.count}}</h1>
    <button m-on:click="increment">Increment</button>
    <router-link to="/">Back Home</router-link>
  </div>`,
  store: store,
  methods: {
    increment: function() {
      store.dispatch("increment", 1);
    }
  }
});

const app = new Moon({
  el: "#app",
  router: router
});
```

**Comment by KingPixil** [comment]
Thanks for reporting this! I've looked into it, and found out that this is an issue with Monx. See, when Moon Router replaces the current component, it is destroyed. 

Monx has a reference to this destroyed component. When the route is changed back, Moon Router replaces the component again, mounting a _new instance_. Monx doesn't have a reference to this instance, and when it attempts to update, it cannot find an element.

I'll try fixing this today in a patch of Monx.

**Comment by KingPixil** [comment]
Can you update Monx to `v0.1.0` and tell me if this issue persists? It should have been fixed in KingPixil/monx@5a91322

**Comment by rogierverbrugge** [comment]
Fixed!
